### ISSUE TITLE
[a11y]: Add aria-label support in Accordion buttons

### ISSUE THREAD

**Issue by srikant-ch5** [carbon-design-system/carbon#19314]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.71.0

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

We have a usecase where we have to show narrow panel with icons and no text in the accordion items.
When we scan using the checker tool it throws below violation.

<img width="840" alt="Image" src="https://github.com/user-attachments/assets/adeadf5b-93b1-46c8-8076-6746e261e32b" />

When we add `aria-label` in `AccordionItem` to fix above violation it is added to `li` element and not the `button` which causes below additional violation.

<img width="889" alt="Image" src="https://github.com/user-attachments/assets/cd3234e3-f77f-4a9f-af7c-6d36e96ea672" />

<img width="556" alt="Image" src="https://github.com/user-attachments/assets/cbcaa772-ac3e-4c51-82e1-9cbcc4b3a5e7" />

### Reproduction/example

https://elyra-canvas-test-harness.u20youmx4sm.us-south.codeengine.appdomain.cloud/#/

### Steps to reproduce

1. Open https://elyra-canvas-test-harness.u20youmx4sm.us-south.codeengine.appdomain.cloud/#/
2. Click on Common Canvas Icon on top right

<img width="341" alt="Image" src="https://github.com/user-attachments/assets/532df468-b12b-44bd-8d93-d8e9f9f49a2c" />
3. Select Flows example from right panel

<img width="301" alt="Image" src="https://github.com/user-attachments/assets/2c977fb9-6949-4285-91dd-926cb9e09912" />

4. Shrink palette using palette resize button

<img width="124" alt="Image" src="https://github.com/user-attachments/assets/7f23324a-e7f4-448e-86b1-c4a7ae9a1e40" />

 

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by srikant-ch5** [comment]
Hi @Gururajj77 could you please let me know ETA for this issue ? Thanks.

**Comment by Gururajj77** [comment]
Hey @srikant-ch5 , you could use the UI Shell's side nav component to implement the same behaviour with accessibility compliance:
https://react.carbondesignsystem.com/?path=/story/components-ui-shell-sidenav--side-nav-rail-w-header
